### PR TITLE
CD-148455 Fix spawn logic. It was preventing spawn ups

### PR DESCRIPTION
**Pull request by johnny-lai**
- [x] @supriya 
- [x] @edk

### CODE REVIEW

**Comment by edk** [comment]
👍

File: lib/resque/pool.rb [hunk 1]
```diff
@@ -441,9 +441,9 @@ def worker_delta_for(queues)
 
       # Only allow downwards deltas while spawn is limited
       ql = spawn_limiter[queues]
-      if delta > 0 && ql.should_spawn?
+      if ql.should_spawn?
         delta
-      else
+      elsif delta > 0
```
Comment (supriya) [inline lib/resque/pool.rb]: if delta > 0, we return 0 as delta?

Comment (johnny-lai) [reply lib/resque/pool.rb]: So `delta = config.fetch(queues, 0) - workers.fetch(queues, []).size`

So when delta is > 0, it wants to spawn. But if `ql.should_spawn?` is `false`, then we should return `0` instead.

If `delta <= 0`, then we allow that regardless of `ql.should_spawn?`. So the if still looks wrong ||^^

**Comment by supriya** [comment]
lgtm

**Comment by johnny-lai** [comment]
[lockit v4.9.0] merged via Lockit CLI on 2019-03-29 13:49:35.348923 -0700 PDT m=+5.291827924

**Pull request merged**